### PR TITLE
Code_800E8EA0 Docs

### PR DESCRIPTION
**Pull request by hensldm**
Documents `Code_800E8EA0` which has functions similar to `SubS` that are used to rotate a player's model towards different points (generally player).

### CODE REVIEW

File: src/code/z_play.c [hunk 1]
```diff
@@ -312,7 +312,7 @@ s32 Play_IsUnderwater(GlobalContext* globalCtx, Vec3f* pos) {
 // z_demo and EnTest4
 // This data appears to be a boolean. It is only set by Play_Init.
 s32 func_8016A168(void) {
-    return D_801D0D50;
+    return gDbgCamEnabled;
 }
```
Comment (EllipticEllipsis) [inline src/code/z_play.c:316]: Looks like this can be named now.

File: src/code/code_800E8EA0.c [hunk 2]
```diff
@@ -1,107 +1,138 @@
...
 
-s32 nop_800E8EEC(UNK_TYPE4 param_1) {
-    return 0;
+s32 Flags_GetInfTable(s32 flag) {
+    return false;
 }
 
-void nop_800E8EFC(UNK_TYPE4 param_1) {
+void Flags_SetInfTable(s32 flag) {
 }
```
Comment (EllipticEllipsis) [inline src/code/code_800E8EA0.c:20]: What's the justification for naming these like this, if they're unused, and don't have nontrivial content?

Should probably note somewhere that they don't actually do anything, too.

File: src/code/code_800E8EA0.c [hunk 3]
```diff
@@ -1,107 +1,138 @@
...
+s32 Actor_RotateForward(Vec3s* headRot, Vec3s* torsoRot) {
+    Math_SmoothStepToS(&headRot->y, 0, 6, 0x1838, 0x64);
+    Math_SmoothStepToS(&headRot->x, 0, 6, 0x1838, 0x64);
+    Math_SmoothStepToS(&torsoRot->y, 0, 6, 0x1838, 0x64);
+    Math_SmoothStepToS(&torsoRot->x, 0, 6, 0x1838, 0x64);
+    return true;
 }
 
-s32 func_800E8FA4(Actor* actor, Vec3f* param_2, Vec3s* param_3, Vec3s* param_4) {
+s32 Actor_RotateToPoint(Actor* actor, Vec3f* target, Vec3s* headRot, Vec3s* torsoRot) {
```
Comment (EllipticEllipsis) [inline src/code/code_800E8EA0.c:30]: Any reason this one didn't get documentation like the other two?

File: src/code/code_800E8EA0.c [hunk 4]
```diff
@@ -1,107 +1,138 @@
...
+ * @param[in,out] headRot the computed actors' head's rotation step
+ * @param[in,out] torsoRot the computed actors' torso's rotation step
+ * @param[in] focusPos the point to set as the actor's focus position
+ *
+ * @return s32 boolean, true if rotated towards player, false if roatated back forwards.
+ *
+ * @note if in a cutscene or debug camera is enabled, the computed rotation will turn towards the view eye no matter the
+ * yaw.
+ */
+s32 Actor_TurnToPlayerSetFocus(GlobalContext* globalCtx, Actor* actor, Vec3s* headRot, Vec3s* torsoRot,
```
Comment (EllipticEllipsis) [inline src/code/code_800E8EA0.c:113]: It would be good to emphasise more in the comments what the difference is between these two.

**Review by EllipticEllipsis — COMMENTED** [review]
Nice job, a few comments. Also, yet another "miscellaneous stuff" file?

(Note: this is all one file due to the padding or lack thereof and the function addresses: seems very unlikely the possible split options are actually splits, quite apart from what other versions do.)

File: src/code/code_800E8EA0.c (hunk 2 above)
Comment (hensldm) [reply src/code/code_800E8EA0.c:20]: I'll admit nothing definitive, but my reasoning is based on `OoT`.
Just below `func_80035B18` (MM's equivalent is `func_800E8EA0`/`Actor_ContinueText`) are these four functions with the same function definitions, in the same order, and later on, after the more substantial flags functions, appears the rotation functions.

File: src/code/code_800E8EA0.c (hunk 3 above)
Comment (hensldm) [reply src/code/code_800E8EA0.c:30]: Didn't think it was needed, as it seems self explanatory.

File: src/code/code_800E8EA0.c (hunk 4 above)
Comment (hensldm) [reply src/code/code_800E8EA0.c:113]: Added another sentence to each to help explain.

File: src/code/code_800E8EA0.c [hunk 5]
```diff
@@ -1,107 +1,152 @@
...
+/**
+ * InfTable does not exist, so this always returns false
+ */
+s32 Flags_GetInfTable(s32 flag) {
+    return false;
 }
 
-void nop_800E8EFC(UNK_TYPE4 param_1) {
+/**
+ * InfTabe does not exist, so this does nothing
```
Comment (engineer124) [inline src/code/code_800E8EA0.c:29]:
```suggestion
 * InfTable does not exist, so this does nothing
```

**Review by EllipticEllipsis — APPROVED** [review]

**Comment by hensldm** [comment]
Okay updated the PR with the OoT discussion, as well as expanded the PR slightly to update the SubS terminology to match these similar functions.

File: src/code/z_play.c [hunk 6]
```diff
@@ -311,8 +311,8 @@ s32 Play_IsUnderwater(GlobalContext* globalCtx, Vec3f* pos) {
 
 // z_demo and EnTest4
 // This data appears to be a boolean. It is only set by Play_Init.
-s32 func_8016A168(void) {
-    return D_801D0D50;
+s32 Play_IsDebugCamEnabled(void) {
+    return gDbgCamEnabled;
 }
```
Comment (engineer124) [inline src/code/z_play.c:316]: These comments above this function can go now that it's named imo

**Review by EllipticEllipsis — APPROVED** [review]

File: include/z_en_hy_code.h [hunk 7]
```diff
@@ -57,10 +57,10 @@ typedef struct EnHy {
...
     /* 0x206 */ Vec3s jointTable[ENHY_LIMB_MAX];
     /* 0x266 */ Vec3s morphTable[ENHY_LIMB_MAX];
-    /* 0x2C6 */ Vec3s turnTarget;
+    /* 0x2C6 */ Vec3s trackTarget;
     /* 0x2CC */ Vec3s headRot;
     /* 0x2D2 */ Vec3s torsoRot;
-    /* 0x2D8 */ Vec3s tmpTurnTarget;
+    /* 0x2D8 */ Vec3s tmptrackTarget;
     /* 0x2DE */ Vec3s tmpHeadRot;
     /* 0x2E4 */ Vec3s tmpTorsoRot;
```
Comment (engineer124) [inline include/z_en_hy_code.h:65]: This is probably related to a different `SubS` PR I missed, but what are these `tmp` prefixes for?

Comment (hensldm) [reply include/z_en_hy_code.h:65]: Ah, those don't really have anything to do with `SubS`. Instead the `EnHy` system has space to backup the three variables above without the `tmp` prefix, and restore them back (same with `tmpActionFunc`). `EnCne01` and `EnBba01` uses it save these three variables just before talking, and restore them back after finishing talking. Probably should rename those better (kind of thinking suffixing `backup`), but would prefer to do that in a different PR as I've used the same terminology in other places that I would also like to change.

Comment (engineer124) [reply include/z_en_hy_code.h:65]: Ah fair, I believe the prefix `prev` is fairly common in OoT/MM for that kind of stored temp with the intent to restore. But that can be a separate PR yeah

**Review by engineer124 — APPROVED** [review]

File: src/code/code_800E8EA0.c [hunk 8]
```diff
@@ -1,107 +1,148 @@
...
-    Math_SmoothStepToS(&param_1->x, 0, 6, 6200, 100);
-    Math_SmoothStepToS(&param_2->y, 0, 6, 6200, 100);
-    Math_SmoothStepToS(&param_2->x, 0, 6, 6200, 100);
-    return 1;
+s32 Actor_TrackNone(Vec3s* headRot, Vec3s* torsoRot) {
+    Math_SmoothStepToS(&headRot->y, 0, 6, 0x1838, 0x64);
+    Math_SmoothStepToS(&headRot->x, 0, 6, 0x1838, 0x64);
+    Math_SmoothStepToS(&torsoRot->y, 0, 6, 0x1838, 0x64);
+    Math_SmoothStepToS(&torsoRot->x, 0, 6, 0x1838, 0x64);
+    return true;
```
Comment (isghj5) [inline src/code/code_800E8EA0.c:39]: Wait... this is a duplicate of code found in actors... was that actor code worthless then? Did they write this later..?

https://github.com/zeldaret/mm/blob/8049dec3393c4ade687dbdfd84d3ccde73168ab0/src/overlays/actors/ovl_En_Ani/z_en_ani.c#L297-L301

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**